libraries/stdlib/src/kotlin/jvm/JvmFlagAnnotations.kt

```diff
@@ -19,18 +19,39 @@ package kotlin.jvm
 import java.lang.annotation.Retention
 import java.lang.annotation.RetentionPolicy
 
+/**
+ * Marks the JVM backing field of the annotated property as `volatile`, meaning that it can
+ * be modified asynchronously by concurrently running threads.
```

Comment:
Made an attempt. This is in any case not a suitable place for explaining the Java memory model in detail.